### PR TITLE
Refactor Services to remove dummy proxy

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

### CODE REVIEW

File: pilot/pkg/model/push_context.go [hunk 1]
```diff
@@ -766,24 +766,19 @@ func (ps *PushContext) GatewayServices(proxy *Proxy) []*Service {
...
-func (ps *PushContext) Services(proxy *Proxy) []*Service {
-	// If proxy has a sidecar scope that is user supplied, then get the services from the sidecar scope
-	// sidecarScope.config is nil if there is no sidecar scope for the namespace
-	if proxy != nil && proxy.SidecarScope != nil && proxy.Type == SidecarProxy {
-		return proxy.SidecarScope.services
-	}
-
+// servicesExportToNamespace returns the list of services that are visible to a namespace.
+// namespace "" indicates all namespaces
+func (ps *PushContext) servicesExportToNamespace(ns string) []*Service {
```
Comment (ramaraochavali) [inline pilot/pkg/model/push_context.go:771]: nit: servicesExportToNamespace -> servicesExportedToNamespace

Comment (ramaraochavali) [reply pilot/pkg/model/push_context.go:771]: why do not we just use the existing Services method and change it arg to take namespace?

**Review by ramaraochavali — COMMENTED** [review]
LGTM other than one naming nit. Will leave it open for @howardjohn to review.

File: pilot/pkg/model/push_context.go (hunk 1 above)
Comment (howardjohn) [reply pilot/pkg/model/push_context.go:771]: It should at least be private

**Review by howardjohn — APPROVED** [review]
Much better!

**Pull request merged**